### ISSUE TITLE
Enhancement: LangSmith support

### ISSUE THREAD

**Issue by mhorijp** [danny-avila/LibreChat#839]
### What features would you like to see added?

I want to use LangSmith with LibreChat.

LangSmith is a new developer platform for a new type of application.
Especially I want to use monitoring feature.

https://www.langchain.com/langsmith

### More details

Set up instruction described in LangSmith tool is below.

 * Install langchain
 
pip install -U langchain

 * Configure environment to connect to LangSmith.

export LANGCHAIN_TRACING_V2=true
export LANGCHAIN_ENDPOINT="https://api.smith.langchain.com"
export LANGCHAIN_API_KEY="<your-api-key>"
export LANGCHAIN_PROJECT="demo"

 * Run any LLM, Chat model, or Chain. Its trace will be sent to this project.
from langchain.chat_models import ChatOpenAI

llm = ChatOpenAI()
llm.predict("Hello, world!")

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
Sorry I'm just getting around to this. I think this should just be as simple as configuring the variables right?

**Comment by danny-avila** [comment]
from what I understand, you just have to set the langsmith environment variables in your .env file, and langchain should pick them up without any edits to code

```bash
LANGCHAIN_TRACING_V2=true
LANGCHAIN_PROJECT=Tracing Walkthrough - {unique_id}
LANGCHAIN_ENDPOINT=https://api.smith.langchain.com
LANGCHAIN_API_KEY=<YOUR-API-KEY>
```

source: https://python.langchain.com/docs/langsmith/walkthrough

**Comment by mariuskolleck** [comment]
> and, you just have to set the langsmith environment variables in your .env file, and langchain should pick them up without any edits to code

Can anyone confirm this works? I cant't get librechat to send any traces to LangSmith. I feel like i'm started brute-forcing it and i'm in need for some guidance 😅

Here is what I have done so far:

- Created LangSmith Account, generated API Key

- I confirmed it worked in general by setting up a very basic langchain project and enabled tracing via env variables

- I added the environment variables to the .env file as mentioned by @danny-avila, then rebuild my docker-compose and sent some chat messages, uploaded a document, asked questions about it --> no success

- I added the environment variables to the compose.override.yml file for api, as well as rag-api container services, rebuilt docker-compose and verified via docker inspect the env varibles are set. Redid my tests --> still no success

- I realized there are also somehow environment variables in the play that are named e.g. "LANGSMITH_TRACING", "LANGSMITH_API_KEY". So I added all those as well, rebuild, retested --> still no success

- Then i started to add decorators to the rag-api main.py's routes as mentioned here:
https://docs.smith.langchain.com/observability

This is my docker-compose.override.yml:

![Image](https://github.com/user-attachments/assets/be81995c-17d9-4737-b31b-d1626dd876f9)
![Image](https://github.com/user-attachments/assets/3885469c-2597-4bbe-82e0-d2e939c1f71a)
